cli/src/build/onboarding/android/gcp-api.ts

```
@@ -0,0 +1,382 @@
+ *  - allowed chars: letters, digits, space, hyphen (`-`), apostrophe (`'`),
+ *    exclamation (`!`), period (`.`)
+ *  - must start and end with a letter or digit
+ *
+ * We strip any disallowed character (including em-dashes — which break the
+ * CLI's placeholder string literals if not handled here), collapse runs of
+ * whitespace, and trim the ends. Falls back to `"Capgo Build"` when the
+ * sanitized result would be shorter than 4 chars.
+ */
+export function sanitizeGcpProjectDisplayName(input: string): string {
+  const fallback = 'Capgo Build'
+  const allowed = input.replace(/[^A-Za-z0-9 \-'!.]/g, ' ').replace(/\s+/g, ' ').trim()
```

ESLint

[error] The character class(es) '[^A-Za-z0-9 \-'!.]' can be simplified using the `i` flag.

(regexp/use-ignore-case)